realm/realm-library/src/androidTest/java/io/realm/NotificationsTest.java

```diff
@@ -686,38 +687,30 @@ public void run() {
                 RealmChangeListener<Realm> listener = new RealmChangeListener<Realm>() {
                     @Override
                     public void onChange(Realm object) {
-                        try {
-                            fail("This handler should not be notified");
-                        } catch (AssertionFailedError e) {
-                            threadAssertionError[0] = e;
-                            handlerNotified.countDown(); // Make sure that that await() doesn't fail instead.
-                        }
+                        fail("This handler should not be notified");
```

Comment:
Yes, verified, with `@RunWith(AndroidJUnit4.class)` it would just fail in the background thread. The behaviour you mentioned is the old android test behaviour.